python/ray/_raylet.pyx

```diff
@@ -640,10 +640,14 @@ cdef int prepare_resources(
         if value < 0:
             raise ValueError("Resource quantities may not be negative.")
         if value > 0:
+            unit_resources = f"{RayConfig.instance().predefined_unit_instance_resources()},\
+            {RayConfig.instance().custom_unit_instance_resources()}"
```

Comment:
lets split by (,) and convert them to a set and then check. Otherwise, we may mismatch: e.g. we have a unit resource called Foo_Bar and you will think Foo is also unit resource.